### ISSUE TITLE
AreaDetector Setup is Confusing and Unintuitive

### ISSUE THREAD

**Issue by ZLLentz** [pcdshub/pcdsdevices#220]
From a discussion I had with @koglin.
The way that we're doing area detector here is very confusing and hard to follow. We have specific `DetectorBase` subclasses that do the same thing (adapt to our PV conventions) and what we use in the `PIM` object is a holdover from skywalker where very specific things are emphasized over everything else.

I suggest the following fixes:
- areadetector/detectors.py should have an `AreaDetector` class instead of `PulnixDetector` and `FeeOpalDetector` and a `DefaultAreaDetector` class with sane default plugins
- areadetector/cam.py should be cleaned up and looked over, it is tough to follow. This isn't being used anywhere, but you could imagine that these sorts of classes would be useful.
- areadetector/plugins.py should be cleaned up and looked over, we should see how many of our overrides made it into the `ophyd` library
- `PIM` should switch to using `DefaultAreaDetector` instead of having a special `PIMPulnixDetector`.

**Comment by teddyrendahl** [comment]
Grabbing this one!

- [ ] Support for `AreaDetector Stream`
- [ ] Remove `areadetector` support from `PIM`
- [ ] Deprecate existing plugins
